src/core/thread/dua_manager.cpp

```diff
@@ -184,6 +242,498 @@ otError DuaManager::Store(void)
+    {
+        dua = GetDomainUnicastAddress();
+        SuccessOrExit(error = Tlv::AppendTlv(*message, ThreadTlv::kTarget, &dua, sizeof(dua)));
+        SuccessOrExit(error = Tlv::AppendTlv(*message, ThreadTlv::kMeshLocalEid, &mle.GetMeshLocal64().GetIid(),
+                                             sizeof(Ip6::InterfaceIdentifier)));
+        mDuaState             = kRegistering;
+        mLastRegistrationTime = TimerMilli::GetNow();
+    }
+    else
+#endif // OPENTHREAD_CONFIG_DUA_ENABLE
+    {
+#if OPENTHREAD_CONFIG_TMF_PROXY_DUA_ENABLE
+        if (!mRegisterCurrentChildIndex)
+        {
+            for (Child &child : Get<ChildTable>().Iterate(Child::kInStateValid))
+            {
+                uint16_t childIndex = Get<ChildTable>().GetChildIndex(child);
+
+                if (mChildDuaMask.Get(childIndex) && !mChildDuaRegisteredMask.Get(childIndex))
+                {
+                    mChildIndexDuaRegistering = childIndex;
+                    break;
+                }
+            }
+        }
+        {
+            uint32_t            lastTransactionTime;
+            const Ip6::Address *duaPtr = nullptr;
+            Child *             child  = Get<ChildTable>().GetChildAtIndex(mChildIndexDuaRegistering);
+
+            VerifyOrExit((duaPtr = child->GetDomainUnicastAddress()) != nullptr, error = OT_ERROR_ABORT);
+
+            dua                 = *duaPtr;
+            lastTransactionTime = Time::MsecToSec(TimerMilli::GetNow() - child->GetLastHeard());
+
+            SuccessOrExit(error = Tlv::AppendTlv(*message, ThreadTlv::kTarget, &dua, sizeof(dua)));
+            SuccessOrExit(error = Tlv::AppendTlv(*message, ThreadTlv::kMeshLocalEid, &child->GetMeshLocalIid(),
+                                                 sizeof(Ip6::InterfaceIdentifier)));
+            SuccessOrExit(error = Tlv::AppendUint32Tlv(*message, ThreadTlv::kLastTransactionTime, lastTransactionTime));
+        }
+#endif // OPENTHREAD_CONFIG_TMF_PROXY_DUA_ENABLE
+    }
+
+    if (!mle.IsFullThreadDevice() && mle.GetParent().IsThreadVersion1p1())
+    {
+        uint8_t pbbrServiceId;
+
+        SuccessOrExit(error = Get<BackboneRouter::Leader>().GetServiceId(pbbrServiceId));
+        SuccessOrExit(error = mle.GetServiceAloc(pbbrServiceId, messageInfo.GetPeerAddr()));
+    }
+    else
+    {
+        messageInfo.GetPeerAddr().SetToRoutingLocator(mle.GetMeshLocalPrefix(),
+                                                      Get<BackboneRouter::Leader>().GetServer16());
+    }
+
+    messageInfo.SetPeerPort(kCoapUdpPort);
+    messageInfo.SetSockAddr(Get<Mle::MleRouter>().GetMeshLocal16());
+
+    SuccessOrExit(error = Get<Coap::Coap>().SendMessage(*message, messageInfo, &DuaManager::HandleDuaResponse, this));
+
+    mIsDuaPending = true;
+
+    // TODO: (DUA) need update when CSL is enabled.
+    if (!Get<Mle::Mle>().IsRxOnWhenIdle())
+    {
+        Get<DataPollSender>().SendFastPolls();
+    }
+
+exit:
+    if (error == OT_ERROR_NO_BUFS)
```

Comment:
which function-call will return this error? The message creation:
```c++
// Prepare DUA.req
    VerifyOrExit((message = Get<Coap::Coap>().NewPriorityMessage()) != nullptr, error = OT_ERROR_NO_BUFS);
```
also produces this error, has this been taken into account?

Reply:
This and following `AppendTlv()` may cause `OT_ERROR_NO_BUFS` error, when it happens, here simply wait some time and then try again.

How should this be taken care in your mind?

Reply:
I believe other functions only return `OT_ERROR_NO_BUFS` theoretically, but actually it never happens.

Reply:
@librasungirl just to make sure that it is by intentional. Clear to me now 👍